src/pam_setquota.rs

```diff
@@ -0,0 +1,159 @@
+#![feature(libc)]
+#![allow(unused_variables)]
+extern crate libc;
+#[macro_use] extern crate mdo;
+extern crate nix;
+extern crate users;
+extern crate pam;
+extern crate mnt;
+#[macro_use] extern crate nom;
+extern crate syslog;
+
+use libc::{c_char, c_int};
+use pam::module; // https://tozny.github.io/rust-pam/pam/module/index.html
+use pam::constants::*;
+use syslog::{Facility,Severity};
+use std::borrow::Cow;
+use nix::sys::quota::quota;
+
+
+#[no_mangle]
+pub extern fn pam_sm_open_session(pamh: &module::PamHandleT, flags: PamFlag,
+                                  argc: c_int, argv: *mut *const c_char
+) -> PamResultCode {
+    use mdo::result::{bind,ret};
+    use users::os::unix::UserExt;
+    use mnt::get_mount;
+    use nix::sys::quota::quotactl_set;
+
+    let args = unsafe { translate_args(argc, argv) };
+
+    // We use the Result<_, pam::constant> monad:
+    //  if something triggers an error, it has an associated pam::constant value
+    //  and processing stop (we go to .unwrap_or_else() and log the error)
+    (mdo! {
```

Comment:
Because the value of the `mdo!` expression is used in the `unwrap_or_else` call, whereas for the other one, the value is simply ignored (there is nothing to be done if an error occurs while trying to log an error...)